nautobot/core/utilities.py

```diff
@@ -29,16 +29,52 @@ def check_filter_for_display(filters, field_name, values):
     if field_name not in filters.keys():
         return resolved_filter
 
-    resolved_filter["display"] = filters[field_name].label or field_name
-    if len(values) == 0 or not hasattr(filters[field_name], "queryset") or not is_uuid(values[0]):
+    filter_field = filters[field_name]
+
+    resolved_filter["display"] = get_filter_field_label(filter_field)
+
+    if len(values) == 0 or not hasattr(filter_field, "queryset") or not is_uuid(values[0]):
         return resolved_filter
     else:
         try:
             new_values = []
-            for value in filters[field_name].queryset.filter(pk__in=values):
+            for value in filter_field.queryset.filter(pk__in=values):
                 new_values.append({"name": str(value.pk), "display": getattr(value, "display", str(value))})
             resolved_filter["values"] = new_values
         except (FieldError, AttributeError):
             pass
 
     return resolved_filter
+
+
+def get_filter_field_label(filter_field):
+    """
+    Return a label for a given field name and value.
+
+    Args:
+        field (Filter): The filter to get a label for
+
+    Returns:
+        (str): The label for the given field
+    """
+    from nautobot.extras.filters.mixins import RelationshipFilter
+    from nautobot.extras.filters.customfields import CustomFieldFilterMixin
+
+    if filter_field.label:
+        return filter_field.label
+    elif isinstance(filter_field, RelationshipFilter):
+        return filter_field.relationship.get_label(side=filter_field.side)
+    elif isinstance(filter_field, CustomFieldFilterMixin):
+        return filter_field.custom_field.label
+    else:
+        return field_name_to_display(filter_field.field_name)
+
+
+def field_name_to_display(field_name):
```

Comment:
Possibly prepend a `_` to this function to mark it as private? Should always get here by going through `get_filter_field_label` rather than calling this function directly, right?

Reply:
Yeah at the moment I don't see other needs to use it.